File: front/src/types/character.ts
export type CharacterConfig = {
    general: {
        id: string;
        name: string;
    };
    multiVoice: boolean;
    voice: {
        name: string;
        type: "voicevox"| "bertvits2" | "irodori-tts" | "stylebertvits2" | "nijivoice" | "google-text-to-speech" | "openai-speech";
        image: string;
        identification: string;
        modelId: string;
        modelFile: string;
        speakerId: string;
        referenceAudioPath: string;
        backgroundImagePath: string;
        behavior: {
            identification: string;
            imagePath: string;
        }[];
    }[];
    chat: {
        type: string;
        model: string;
        systemPrompt: string;
        temperature: {
            enable: boolean;
            value: number;
        }
        maxHistory: number;
        limit:{
            day: {
                request: number;
                token: number;
            },
            hour: {
                request: number;
                token: number;
            },
            minute: {
                request: number;
                token: number;
            }
        }
    };
    memory: {
        enabled: boolean;
        maxItemsInPrompt: number;
        enableRelationshipMemory: boolean;
        enableSessionSummary: boolean;
        enableSemanticSearch: boolean;
        embeddingModel: string;
        allowSensitiveMemory: boolean;
    };
};

export type CharacterConfigList = {
    characters: CharacterConfig[];
};

export type MemoryItem = {
    id: string;
    characterId: string;
    ownerId: string;
    scope: "character" | "relationship";
    kind: string;
    content: string;
    keywordsText: string;
    pinned: boolean;
    confidence: number;
    salience: number;
    source: string;
    updatedAt: string;
};

export type MemoryItemList = {
    items: MemoryItem[];
};

export type SessionSummary = {
    ownerId: string;
    characterId: string;
    sessionId: string;
    summary: string;
    updatedAt: string;
};

File: server/data/character.go
package data

type CharacterConfigList struct {
	Characters []CharacterConfig `json:"characters"`
}

type CharacterConfig struct {
	General    CharacterConfigGeneral `json:"general"`
	MultiVoice bool                   `json:"multiVoice"`
	Voice      []CharacterConfigVoice `json:"voice"`
	Chat       CharacterConfigChat    `json:"chat"`
	Memory     CharacterConfigMemory  `json:"memory"`
}

type CharacterConfigGeneral struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

type CharacterConfigVoice struct {
	Name                string                         `json:"name"`
	Type                string                         `json:"type"`
	Identification      string                         `json:"identification"`
	ModelID             string                         `json:"modelId"`
	ModelFile           string                         `json:"modelFile"`
	SpeakerID           string                         `json:"speakerId"`
	ReferenceAudioPath  string                         `json:"referenceAudioPath"`
	Image               string                         `json:"image"`
	BackgroundImagePath string                         `json:"backgroundImagePath"`
	Behavior            []CharacterConfigVoiceBehavior `json:"behavior"`
}

type CharacterConfigVoiceBehavior struct {
	Identification string `json:"identification"`
	ImagePath      string `json:"imagePath"`
}

type CharacterConfigChat struct {
	Type         string                   `json:"type"`
	Model        string                   `json:"model"`
	SystemPrompt string                   `json:"systemPrompt"`
	Temperature  TemperatureConfig        `json:"temperature"`
	MaxHistory   int64                    `json:"maxHistory"`
	Limit        CharacterConfigChatLimit `json:"limit"`
}

type CharacterConfigMemory struct {
	Enabled                  bool   `json:"enabled"`
	MaxItemsInPrompt         int64  `json:"maxItemsInPrompt"`
	EnableRelationshipMemory bool   `json:"enableRelationshipMemory"`
	EnableSessionSummary     bool   `json:"enableSessionSummary"`
	EnableSemanticSearch     bool   `json:"enableSemanticSearch"`
	EmbeddingModel           string `json:"embeddingModel"`
	AllowSensitiveMemory     bool   `json:"allowSensitiveMemory"`
}

type TemperatureConfig struct {
	Enable bool    `json:"enable"`
	Value  float32 `json:"value"`
}

type CharacterConfigChatLimit struct {
	Day    CharacterConfigChatLimitType `json:"day"`
	Hour   CharacterConfigChatLimitType `json:"hour"`
	Minute CharacterConfigChatLimitType `json:"minute"`
}

type CharacterConfigChatLimitType struct {
	Request int64 `json:"request"`
	Token   int64 `json:"token"`
}
